cmd/gdsutil/main.go

```diff
@@ -1244,6 +1257,83 @@ func generateTokenKey(c *cli.Context) (err error) {
 	return nil
 }
 
+func migrateContacts(c *cli.Context) (err error) {
+	// Create a list of vasp contacts and model contacts
+	modelContacts := make(map[string]*models.Contact)
+	var vaspContacts []*pb.Contact
+
+	// iterate through all vasps in the database
+	iter := ldb.NewIterator(util.BytesPrefix([]byte(wire.NamespaceVASPs)), nil)
+	for iter.Next() {
+		vasp := new(pb.VASP)
+		if err = proto.Unmarshal(iter.Value(), vasp); err != nil {
+			iter.Release()
+			return cli.Exit(err, 1)
+		}
+		iter := models.NewContactIterator(vasp.Contacts, true, false)
+
+		// Iterate through all contacts on the vasp
+		for iter.Next() {
+			vaspContact, _ := iter.Value()
+			vaspContacts = append(vaspContacts, vaspContact)
+			modelContact, AlreadyExists := modelContacts[vaspContact.Email]
+
+			// Extract the extra fields from the vasp contact
+			vaspContactExtra := &models.GDSContactExtraData{}
+			if vaspContact.Extra != nil {
+				if err = vaspContact.Extra.UnmarshalTo(vaspContactExtra); err != nil {
+					return fmt.Errorf("could not deserialize previous extra: %s", err)
+				}
+			}
+
+			// If the model contact doesn't already exist create it,
+			// if it does update the email log and vasp list
+			if !AlreadyExists {
+				modelContact = &models.Contact{
+					Email:    vaspContact.Email,
+					Name:     vaspContact.Name,
+					Vasps:    []string{vasp.CommonName},
+					Verified: vaspContactExtra.Verified,
+					Token:    vaspContactExtra.Token,
+					EmailLog: vaspContactExtra.EmailLog,
+					Created:  time.Now().Format(time.RFC3339),
+					Modified: time.Now().Format(time.RFC3339),
+				}
+			} else {
+				modelContact.Vasps = append(modelContact.Vasps, vasp.CommonName)
+				modelContact.EmailLog = append(modelContact.EmailLog, vaspContactExtra.EmailLog...)
+			}
+		}
+	}
+
+	// if the dryrun flag is set, print all vasp contacts and the
+	// newly created model contacts and return
+	if c.Bool("dryrun") {
+		fmt.Println("existing vasp contacts:")
+		for _, contact := range vaspContacts {
+			fmt.Print(contact)
```

Comment:
Might want to newline separate these or something or just print out the emails space separated because this is probably going to be messy when things get printed out.

Reply:
Good Idea!